File: packages/shared/src/validate-snapshot.mjs
import { readFile } from "node:fs/promises";
import { validateDashboardSnapshot } from "./schema.mjs";

const file = process.argv[2];
if (!file) {
  console.error("Usage: node validate-snapshot.mjs <snapshot.json>");
  process.exit(1);
}

const snapshot = JSON.parse(await readFile(file, "utf8"));
validateDashboardSnapshot(snapshot);
console.log(`Snapshot is valid: ${file}`);

File: packages/shared/src/validate_snapshot.py
from pathlib import Path
import json
import sys

sys.path.append(str(Path(__file__).resolve().parent))
from schema import validate_dashboard_snapshot


if len(sys.argv) != 2:
    print("Usage: python3 validate_snapshot.py <snapshot.json>", file=sys.stderr)
    sys.exit(1)

path = Path(sys.argv[1])
snapshot = json.loads(path.read_text(encoding="utf-8"))
validate_dashboard_snapshot(snapshot)
print(f"Snapshot is valid: {path}")

